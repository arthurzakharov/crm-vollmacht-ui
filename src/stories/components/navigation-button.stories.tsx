import { Meta, StoryObj } from "@storybook/react";
import { fn, within, userEvent, expect } from "@storybook/test";
import NavigationButton from "@components/navigation-button";

type Story = StoryObj<typeof NavigationButton>;

const meta: Meta<typeof NavigationButton> = {
  title: "COMPONENTS/NavigationButton",
  component: NavigationButton,
};

export const TypeForward: Story = {
  name: "Type Forward",
  args: {
    type: "forward",
    text: "",
    disabled: false,
    onClick: fn(),
  },
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);
    await step("Click in not disabled state", async () => {
      await userEvent.click(canvas.getByRole("button"));
      await expect(args.onClick).toHaveBeenNthCalledWith(1);
    });
    await step("Button is not in focus after click", async () => {
      await userEvent.click(canvas.getByRole("button"));
      await expect(canvas.getByRole("button")).not.toHaveFocus();
    });
  },
};

export const TypeBackward: Story = {
  name: "Type Backward",
  args: {
    type: "backward",
    text: "",
    disabled: false,
    onClick: fn(),
  },
};

export const WithCustomText: Story = {
  name: "WithCustomText",
  args: {
    type: "forward",
    text: "Some custom text",
    disabled: false,
    onClick: fn(),
  },
};

export const Disabled: Story = {
  name: "Disabled",
  args: {
    type: "forward",
    text: "",
    disabled: true,
    onClick: fn(),
  },
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);
    await step("Click in not disabled state", async () => {
      await userEvent.click(canvas.getByRole("button"));
      await expect(args.onClick).not.toBeCalled();
    });
  },
};

export const DefaultValues: Story = {
  name: "Default Values",
  args: {
    type: "forward",
    text: undefined,
    disabled: undefined,
    onClick: fn(),
  },
};

export default meta;
