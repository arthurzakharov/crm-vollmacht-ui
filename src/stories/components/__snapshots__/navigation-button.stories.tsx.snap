// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS/NavigationButton DefaultValues smoke-test 1`] = `
<button type="button"
        tabindex="0"
        class="navigation-button navigation-button--forward"
>
  Weiter
  <div>
  </div>
</button>
`;

exports[`COMPONENTS/NavigationButton Disabled play-test 1`] = `
<button type="button"
        tabindex="0"
        disabled
        class="navigation-button navigation-button--forward"
>
  Weiter
  <div>
  </div>
</button>
`;

exports[`COMPONENTS/NavigationButton TypeBackward smoke-test 1`] = `
<button type="button"
        tabindex="0"
        class="navigation-button navigation-button--backward"
>
  Zur√ºck
</button>
`;

exports[`COMPONENTS/NavigationButton TypeForward play-test 1`] = `
<button type="button"
        tabindex="0"
        class="navigation-button navigation-button--forward"
>
  Weiter
  <div>
  </div>
</button>
`;

exports[`COMPONENTS/NavigationButton WithCustomText smoke-test 1`] = `
<button type="button"
        tabindex="0"
        class="navigation-button navigation-button--forward"
>
  Some custom text
  <div>
  </div>
</button>
`;
